### ISSUE TITLE
FusedMLP with dropout fails inside Triton during forward pass

### ISSUE THREAD

**Issue by SpirinEgor** [facebookresearch/xformers#713]
# 🐛 Bug

After upgrading to Torch 2.0.0 and XFormers 0.0.17, FusedMLP forward pass fails with
```bash
python: /opt/conda/conda-bld/torchtriton_1677881345124/work/lib/Dialect/TritonGPU/IR/Dialect.cpp:348: unsigned int mlir::triton::gpu::BlockedEncodingAttr::getElemsPerThread(llvm::ArrayRef<long int>) const: Assertion `rank == sizePerThread.size() && "unexpected rank in BlockedEncodingAttr::getElemsPerThread"' failed.
Aborted (core dumped)
```

I noticed that error only happens when `dim_model` is large values and when dropout non-zero.
All perfectly works for `dim_model=128` and fails for `[256, 512, 768]`

## To Reproduce

1. Install Torch 2.0.0, XFormers 0.0.17 (using pip or conda)
2. Run:
```python
import torch
from xformers.components import Activation
from xformers.components.feedforward import FusedMLP

m = FusedMLP(256, 0.1, Activation("gelu"), 4, bias=True).to("cuda:0")
t = torch.rand(32, 1024, 256, device="cuda:0")
m(t)
```

And after replacing `256` to `128` or dropout probability to `0.0`, code would work just as expected.

## Expected behavior

No errors, computed MLP 😅

## Environment

- PyTorch Version (e.g., 1.0): 2.0.0
- OS (e.g., Linux): Ubuntu 22.04 LTS (x86_64)
- How you installed PyTorch (`conda`, `pip`, source): both `conda` and `pip`
- Build command you used (if compiling from source): -
- Python version: 3.10.9
- CUDA/cuDNN version: 11.7
- GPU models and configuration: A100 80Gb
- Any other relevant information: `triton==2.0.0`, `xformers==0.0.17`

**Comment by gaojun4ever** [comment]
I encountered the same issue while using FusedMLP. My version of XFormers is 0.0.20 and my Triton version is 2.0.0. Is there any solution to this problem?
